src/libmongoc/tests/test-mongoc-uri.c

```diff
@@ -255,6 +255,15 @@ test_mongoc_uri_new (void)
    ASSERT (!mongoc_uri_new (
       "mongodb://localhost/?" MONGOC_URI_HEARTBEATFREQUENCYMS "=499"));
 
+   /* timeoutMS */
```

Comment:
We should also test non-numeric values throw a warning. An example from `source/uri-options/tests/connection-options.yml` from [this](https://github.com/mongodb/specifications/pull/873/files) PR is
```
        description: "Non-numeric timeoutMS causes a warning"
        uri: "mongodb://example.com/?timeoutMS=invalid"
        valid: true
        warning: true
        hosts: ~
        auth: ~
        options: {}
```